### PR TITLE
Updates docs to match netlifycms.org/docs

### PR DESCRIPTION
**Pull request by bdougie**
**- Summary**
Documentation in this repository is out of sync with netlifycms.org. I am experimenting with [Markdown Magic](https://github.com/DavidWells/markdown-magic) to sync docs from this repo to the netlify-cms-www repo. See the code [here](https://github.com/netlify/netlify-cms-www/blob/markdown-magic-test/markdown.config.js).

Essentially using comments in Markdown allows the plugin to insert new content during the build step. This will enable contributors to update the docs freely in this repo, but allow us to host the same documentation on netlifycms.org/docs

https://github.com/netlify/netlify-cms-www/pull/20

**- Test plan**

See it live here: https://markdown-magic-test--netlify-cms-www.netlify.com/docs/

**- Description for the changelog**
Updates to documentation to match what is on netlifycms.org

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by erquhart** [comment]
@bdougie +1000 on reading docs from source, really glad to see this!

File: docs/extending.md [hunk 1]
```diff
@@ -1,195 +1,36 @@
...
+## Extending With Widgets
+
+The NetlifyCMS exposes an `window.CMS` global object that you can use to register custom widgets, previews and editor plugins. The available widger extension methods are:
 
-* **registerPreviewStyle** Register a custom stylesheet to use on the preview pane.
-* **registerPreviewTemplate** Registers a template for a collection.
 * **registerWidget** lets you register a custom widget.
 * **registerEditorComponent** lets you add a block component to the Markdown editor
 
-**Writing React Components inline**
```
Comment (erquhart) [inline docs/extending.md]: I think we're losing some important content in this file. Any thoughts @bdougie?

Comment (bdougie) [inline docs/extending.md]: This content was moved to the customization section: https://www.netlifycms.org/docs/customization/

We had a discussion prior to launch and came to the decision that extending the cms with widgets (this section) and customization of the preview pane (new section) are two separate things and now live in two seprate sections.

Comment (erquhart) [inline docs/extending.md]: Dang I checked to see if it moved, didn't look close enough. Thanks.

**Pull request merged**